src/DGtal/geometry/surfaces/ShroudsRegularization.ih

```diff
@@ -0,0 +1,378 @@
+    {
+      double right = 0.0;
+      double  left = 0.0;
+      double  coef = 0.0;
+      const auto s = myPtrIdxSurface->surfel( v );
+      const auto k = myPtrK->sOrthDir( s );
+      for ( Dimension i = 0; i < 3; ++i )
+	{
+	  if ( i == k )  continue; // not a valid slice
+	  const auto   v_i = std::make_pair( v, i );
+	  const auto  vn_i = next( v_i );
+	  const auto vnn_i = next( vn_i );
+	  const auto  vp_i = prev( v_i );
+	  const auto vpp_i = prev( vp_i );
+	  const auto     c = c2_all( v_i );
+	  const auto    cn = c2_all( vn_i );
+	  const auto    cp = c2_all( vp_i );
+	  const RealPoint Xnn = position( vnn_i.first );
+	  const RealPoint  Xn = position( vn_i.first );	    
+	  const RealPoint   X = position( v );
+	  const RealPoint Xpp = position( vpp_i.first );
+	  const RealPoint  Xp = position( vp_i.first );	    
+	  right += beta * ( - get<0>( c ) * get<0>( cn ) * Xnn[ k ]
+			    + ( get<0>( c ) * get<1>( cn )
+				+ get<1>( c ) * get<0>( c ) ) * Xn[ k ]
+			    + ( get<2>( c ) * get<1>( cp )
+				+ get<1>( c ) * get<2>( c ) ) * Xp[ k ]
+			    - get<2>( c ) * get<2>( cp ) * Xpp[ k ] )
+	    + alpha * ( get<0>( c ) * Xn[ k ] + get<2>( c ) * Xp[ k ] ); 
+	  Scalar a = get<0>( c ) * get<2>( cn ) + get<1>( c ) * get<1>( c )
+	    + get<2>( c ) * get<0>( cp );
+	  left  += ( beta * a + alpha * get<1>( c ) ) * myInsV[ v ][ k ];
+	  coef  += ( beta * a + alpha * get<1>( c ) )
+	    * ( myOutV[ v ][ k ] - myInsV[ v ][ k ] );
+	}
+      // Possibly randomization to avoid local minima.
+      newT[ v ] = ( right - left ) / coef
+	+ ( (double) rand() / (double) RAND_MAX - 0.49 ) * randomization; 
+    }
+  auto X     = positions();
+  // Damping between old and new positions.
+  for ( Vertex v = 0; v < myT.size(); ++v )
+    myT[ v ] = 0.5 * newT[ v ] + 0.5 * myT[ v ];
+  enforceBounds();
+  auto Xnext = positions();
+  Scalar  l2 = 0.0;
+  Scalar loo = 0.0;
+  for ( Vertex v = 0; v < myT.size(); ++v ) {
+    loo = std::max( loo, ( Xnext[ v ] - X[ v ] ).norm() );
+    l2 += ( Xnext[ v ] - X[ v ] ).squaredNorm();
+  }
+  return std::make_pair( loo, sqrt( l2 / myT.size() ) );
+}
+
+template < typename TDigitalSurfaceContainer >
+std::pair<double,double>
+DGtal::ShroudsRegularization< TDigitalSurfaceContainer >::
+oneStepSquaredCurvatureMinimization
+( double randomization )
+{
+  parameterize();
+  Scalars newT = myT;
+  for ( Vertex v = 0; v < myT.size(); ++v )
+    {
+      double right = 0.0;
+      double  left = 0.0;
+      double  coef = 0.0;
+      const auto s = myPtrIdxSurface->surfel( v );
+      const auto k = myPtrK->sOrthDir( s );
+      for ( Dimension i = 0; i < 3; ++i )
+	{
+	  if ( i == k )  continue; // not a valid slice
+	  const auto   v_i = std::make_pair( v, i );
+	  const auto  vn_i = next( v_i );
+	  const auto vnn_i = next( vn_i );
+	  const auto  vp_i = prev( v_i );
+	  const auto vpp_i = prev( vp_i );
+	  const auto   c_1 = c1( v_i );
+	  const auto     c = c2_all( v_i );
```

Comment:
FYI: std::tie is a nice stuff to avoid the `get` on tuples or pairs. If `f` returns a tuple:

      double a,b,c,d;
      std::tie(a,b,c,d) = f(....) 

(`get<0>` goes to a, ...)

This makes the code easier to read.

Reply:
Yes I know about it, but (strangely ?) I found it simpler to use get<k>( c ) instead of naming variables. I think I had too many variables already.